lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp

```diff
@@ -633,8 +595,21 @@ bool TypeLoweringVisitor::lowerProducer(
     if (needsSym || op->hasAttr(cache.innerSymAttr)) {
       auto newName = StringAttr::get(context, loweredSymName);
       newOp->setAttr(cache.innerSymAttr, newName);
-      opSymNames[newName] = newOp;
       assert(!loweredSymName.empty());
+
+      // If this operation has an inner symbol, then update the origSymbols
+      // disjoint set to make sure that all derived symbols are associated with
+      // the original symbol.
+      if (innerSymAttr) {
```

Comment:
Just an observation, there is nothing to record in the `innerRefRenames` if the original op didn't have an inner sym, but the lowered op has a symbol(when `needsSym=true`.

Reply:
That should be expected and safe. If the original op didn't have a symbol, but gets one after `LowerTypes` then there can, by definition, be no valid rename.